### PR TITLE
Implement REST endpoint for merging guest cart into customer cart #10

### PR DESCRIPTION
**Pull request by paliarush**
## Description

Solves #10
 

## How Has This Been Tested?

1. Open BrowserA and add several items to the cart as a guest
2. Open BrowserB, login as a customer, add several items to the cart
3. If guest cart ID is not known, open `quote_id_mask` DB table and find corresponding `masked_id`
4. Make `PUT` request to `{{rest_base_url}}/customer-carts/mine/merge-with-guest-cart/<guest_cart_id_goes_here>` endpoint with `Authorization` header containing customer's token
5. Open cart in BrowserA, it should contain all items from both carts
6. Open cart in BrowserB, it should not contain any items

## Known issues

1. If UI flow is combined with REST API flow, as described in the previous section, then guest minicart is not updated properly unless additional action with the guest cart is performed. It is related to the inability to invalidate browser local storage from the backend. This should not be an issue if the whole flow is performed via REST API, which is the case for CIF.
2. Endpoint is not documented in the README
3. No negative scenarios tested
4. Automated tests should be added

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] I ran all tests locally and they pass.

### CODE REVIEW

File: AggregatedServices/etc/webapi.xml [hunk 1]
```diff
@@ -33,4 +33,14 @@
             <parameter name="cartId" force="true">%cart_id%</parameter>
         </data>
     </route>
+
+    <route url="/V1/customer-carts/mine/merge-with-guest-cart/:guestCartId" method="PUT">
```
Comment (masodu) [inline AggregatedServices/etc/webapi.xml]: @paliarush what is the reason to use `customer-carts` and not `carts`? Normally the customer carts endpoint is is just `carts`.

**Review by masodu — COMMENTED** [review]
LGTM, except that I saw that the cart response is different. Can that be reverted?

File: AggregatedServices/etc/webapi.xml (hunk 1 above)
Comment (paliarush) [reply AggregatedServices/etc/webapi.xml]: Agree, can be changed.

**Comment by paliarush** [comment]
@masodu please clarify which response is different and what is the expected response.

**Review by masodu — APPROVED** [review]

File: AggregatedServices/Model/Service/MergeCarts.php [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+            $customerQuote->merge($guestQuote);
+            $customerQuote->setTotalsCollectedFlag(false);
+            $customerQuote->collectTotals();
+            $this->quoteResourceModel->save($customerQuote);
+
+            // Disable guest cart
+            $guestQuote->setIsActive(false);
+            $this->quoteResourceModel->save($guestQuote);
+        } catch (\Exception $e) {
+            throw new LocalizedException(__("Unable to merge specified guest quote into active customer quote."));
```
Comment (melnikovi) [inline AggregatedServices/Model/Service/MergeCarts.php]: Not sure we need to return localized exception.

**Review by melnikovi — APPROVED** [review]

**Review by melnikovi — APPROVED** [review]

**Pull request merged**